pkg/controller/job/job_controller.go

```diff
@@ -772,6 +772,21 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 	}
 
 	activePods := controller.FilterActivePods(pods)
+
+	// As part of KEP-3939,
+	// we will allow for opt-in control of including terminating pods
+	// in our activePod list.
+	// This works because we pass this list into managePods which counts the list of activePods.
+	var terminatingPods *int32 = pointer.Int32(0)
+	if feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) {
+		terminatingPods = controller.CountTerminatingPods(pods)
+	}
+	if recreatePodsWhenFailed(&job) {
+		if pointer.Int32Deref(terminatingPods, 0) > 0 {
+			terminatingPodsList := controller.FilterTerminatingPods(pods)
+			activePods = append(activePods, terminatingPodsList...)
```

Comment:
please change the name of the variable `activePods`, as it's no longer true that these are active pods.

Reply:
@alculquicondor you okay with leaving name as is?

Reply:
Separating the names would help with https://github.com/kubernetes/kubernetes/pull/117015#discussion_r1257541863

Reply:
We've separated into two separate lists.